### PR TITLE
feat: sdk: add `setHeaders` for setting global headers and make sure all client calls have a `headers` arg

### PR DESCRIPTION
**Pull request by onehassan**
resolves https://github.com/nhost/nhost/issues/2696

### CODE REVIEW

**Review by dbarrosop — APPROVED** [review]

File: packages/nhost-js/src/clients/nhost.ts [hunk 1]
```diff
@@ -90,4 +90,20 @@ export class NhostClient {
...
+   *
+   * @example
+   * ```ts
+   * nhost.setRole('admin')
+   * ```
+   *
+   * @docs https://docs.nhost.io/reference/javascript/set-role
+   */
+  setRole(role: string) {
+    this.graphql.setHeaders({ 'x-hasura-role': role })
```
Comment (dbarrosop) [inline packages/nhost-js/src/clients/nhost.ts:105]: what happens if the user does the following?

```
n.setRole("my-role")
n.storage.setHeaders({"x-something": "blah"}) 
```

will the role still be sent? It won't, will it? We should :)

Comment (onehassan) [reply packages/nhost-js/src/clients/nhost.ts:105]: Yes, good catch 👍 - will adjust

File: packages/nhost-js/src/clients/nhost.ts [hunk 2]
```diff
@@ -90,4 +90,38 @@ export class NhostClient {
...
+    this.storage.setHeaders({ 'x-hasura-role': role })
+    this.functions.setHeaders({ 'x-hasura-role': role })
+  }
+
+  /**
```
Comment (dbarrosop) [inline packages/nhost-js/src/clients/nhost.ts:110]: similar comment to above

**Comment by xmlking** [comment]
@onehassan   why do we need to pass argument for unsetRole() function ? `e.g., nhost.unsetRole('admin')`
Is it just removing `x-hasura-role` header?

**Comment by dbarrosop** [comment]
@xmlking copy&paste mistake, will fix

**Pull request merged**